src/walker.js

```diff
@@ -27,32 +27,32 @@ export class WalkerBase {
 	}
 
 	/**
-	 *
-	 * @param {any} parent
-	 * @param {string} prop
-	 * @param {number} index
-	 * @param {BaseNode} node
+	 * @template {Node} Parent
+	 * @param {Parent} parent
+	 * @param {keyof Parent} prop
+	 * @param {number | null} index
+	 * @param {Node} node
 	 */
 	replace(parent, prop, index, node) {
 		if (parent) {
-			if (index !== null) {
-				parent[prop][index] = node;
+			if (index !== null && typeof index !== 'undefined') {
```

Comment:
Personally I prefer the following equivalent:

```suggestion
			if (index != null) {
```

Reply:
Sure, applied